src/App.tsx

```diff
@@ -1,222 +1,234 @@
+
+  const addNewTodoToVisibleTodos = (todo?: Todo, title = 'temp') => {
+    const newTodo = todo || {
+      id: 0,
+      title,
+      completed: false,
+      userId: 0,
+    };
+
+    setTodos((prevTodos) => [...prevTodos, newTodo]);
+  };
+
+  const deleteVisibleTodo = (id: number) => {
+    setTodos((prevTodos) => prevTodos.filter(todo => todo.id !== id));
+  };
+
+  const addIsLoadingTodoId = (...ids: number[]) => {
+    setLoadingTodosId((prevIds) => [...prevIds, ...ids]);
+  };
+
+  const deleteIsLoadingTodos = (...ids: number[]) => {
+    setLoadingTodosId((prevIds) => prevIds.filter((id) => !ids.includes(id)));
+  };
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user.id)
+        .then(todosFromServer => {
+          setTodos(todosFromServer);
+        })
+        .catch(() => showAlert('Unable to get Todos'));
     }
   }, []);
 
+  useEffect(() => {
+    setVisibleTodos(getFilteredTodos());
+  }, [todos, filterTodos]);
+
+  const isInterfaceHidden = (todos.length === 0
+    && filterTodos === TodosFilter.All);
+
+  const handleFilterTodos = (filterValue: TodosFilter) => {
+    setFilterTodos(filterValue);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newTodoTitle = newTodoField.current?.value.trim();
+
+    if (newTodoTitle === '') {
+      handleEmptyFieldError();
+    }
+
+    if (newTodoTitle !== '' && newTodoTitle && user) {
+      clearAlert();
+      setAdding(true);
+      addNewTodoToVisibleTodos(undefined, newTodoTitle);
+      addIsLoadingTodoId(0);
+      createTodo({
+        id: 0,
+        userId: user.id,
+        title: newTodoTitle,
+        completed: false,
+      })
+        .then((todoFromServer) => {
+          addNewTodoToVisibleTodos(todoFromServer);
+          deleteIsLoadingTodos(0);
+        })
+        .catch(() => showAlert('Unable to add Todo'))
+        .finally(() => {
+          if (newTodoField.current) {
+            newTodoField.current.value = '';
+          }
+
+          setAdding(false);
+          deleteVisibleTodo(0);
+        });
+    }
+  };
+
+  const handleChangeStatus = (todoId: number, status: boolean) => {
+    clearAlert();
+    changeTodoStatus(todoId, !status)
+      // .then(() => forceUpdate())
+      .catch(() => showAlert('Unable to update Todo'));
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    clearAlert();
+    addIsLoadingTodoId(todoId);
+    deleteTodo(todoId)
+      .then(() => deleteVisibleTodo(todoId))
+      .catch(() => handleDeleteError())
+      .finally(() => deleteIsLoadingTodos(todoId));
+  };
+
+  const handleToggleAllTodos = () => {
+    const toggleValue = !todos.every((todo) => todo.completed);
+
+    todos.forEach(({ id }) => {
+      clearAlert();
+      changeTodoStatus(id, toggleValue)
+        // .then(() => forceUpdate())
```

Comment:
Uncomment it or delete comment. And let’s use `async await` all over the code. It gives you a simpler way to work with asynchronous promise-based code.

Reply:
Can I leave it like that, and use `async await` in the next tasks? And why it`s simpler?